### PR TITLE
Faster go generate

### PR DESCRIPTION
**Pull request by tamird**
@mrtracy for the ui stuff

### CODE REVIEW

File: proto/Makefile [hunk 1]
```diff
@@ -25,8 +24,8 @@ ENGINE_ROOT := $(REPO_ROOT)/storage/engine
 
 # Ensure we only have one entry in GOPATH
 GOPATH := $(GITHUB_ROOT)/../..
-#                                   ^  ^  ^~ GOPATH
-#                                   |  |~ GOPATH/src
+#                        ^  ^  ^~ GOPATH
+#                        |  |~ GOPATH/src
```
Comment (tschottdorf) [inline proto/Makefile]: any particular reason for the second line?

File: rpc/codec/wire.pb/Makefile [hunk 2]
```diff
@@ -13,17 +13,30 @@
...
+RPC_PROTOS      := $(sort $(wildcard *.proto))
 
-PROTOC      := $(GOPATH)/bin/protoc
-PROTOS      := $(sort $(wildcard *.proto))
-PROTO_GO    := $(PROTOS:.proto=.pb.go)
+RPC_GO_SOURCES  := $(RPC_PROTOS:%.proto=%.pb.go)
 
-PROTO_PATH := ../..:$(GOPATH)/src/github.com/gogo/protobuf:$(GOPATH)/src/github.com/gogo/protobuf/protobuf
-PROTOC_GEN_GOGO := $(GOPATH)/bin/protoc-gen-gogo
+.PHONY:
```
Comment (tschottdorf) [inline rpc/codec/wire.pb/Makefile]: does this do the same as `.PHONY: all`?

File: proto/Makefile (hunk 1 above)
Comment (tamird) [inline proto/Makefile]: a relic of the past. will fix

File: rpc/codec/wire.pb/Makefile (hunk 2 above)
Comment (tamird) [inline rpc/codec/wire.pb/Makefile]: yeah

**Comment by mrtracy** [comment]
LGTM

**Pull request merged**